proxy_agent/src/telemetry/event_reader.rs

```diff
@@ -55,17 +56,29 @@ pub fn start_async(
     _ = thread::Builder::new()
```

Comment:
Same as a comment left earlier, should we use tokio to spawn this thread?

Reply:
I make this function as async.

Reply:
I'd recommend using the `tokio::main` macro which will set up the runtime for the main thread. Then you can get rid of the entire `get_runtime` thing and just spawn the task.